### DISCUSSION TITLE
Change default behaviour not to minimize windows

### DISCUSSION THREAD

**Discussion by samevans-nhsnet** [PSAppDeployToolkit/PSAppDeployToolkit#1165] [Q&A] [answered]
I understand why you may want to minimise windows and I know you can override this behaviour on a case by case basis for each call of an affected cmdlet but it would be great if you could set the "-NoMinimizeWindows" switch as a default for a deployment to avoid the risk it being missed on a command.

The users we support are not going to be happy if we minimise all their windows.

**Comment by mjr4077au** [comment]
This is how it is in v4 because this is how it's always been. Changing defaults is a slippery slope invites conversations from everyone wanting the defautls to be the way they want them.

Have a read of https://devblogs.microsoft.com/powershell-community/how-to-use-psdefaultparametervalues/, it will tell you how to use the `$PSDefaultParameterValues` automatic variable to specify default parameter values in the absence of a bound override.

For `Show-ADTInstallationWelcome`, you could define `-NoMinimizeWindows` in your `$PSDefaultParameterValues` automatic variable inside `Invoke-AppDeployToolkit.ps1` to achieve what you're after.

**Reply by samevans-nhsnet** [reply]
Thank you for your response.  

Sorry, I should have been clearer, I was not expecting you to change the hard-coded default behaviour of the functions themselves (as you point out this sort of change can just create confusion and division unnecessarily).  I was more talking about being able to specify a universal over-ride it in the config file in the same way you can some other function behaviour like the cache folder or File Copy mode.  

The solution you have suggested is a possible work around that I had not considered but adding the option to the config file would be cleaner.

**Reply by samevans-nhsnet** [reply]
Do you have any thoughts on being able to add the default windows minimising behaviour to the config file?

**Reply by mjr4077au** [reply, marked as answer]
@samevans-nhsnet please see https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/pull/1308.
